### ISSUE TITLE
statistic and p-value do not match when show.stat=TRUE and show.std=TRUE

### ISSUE THREAD

**Issue by johannes-titz** [strengejacke/sjPlot#649]
```
data <- Hmisc::spss.get("thomas202006.sav", use.value.labels = F)
thomas <- lmer(liking~emotion+age+emotion:age+(emotion|ID), data = data)
tab_model(thomas, show.stat = T, show.std = T)
```
![model2](https://user-images.githubusercontent.com/32065313/84897010-3ac06080-b0a5-11ea-876b-051bec72ebf2.png)

For variable `Alter` the t-value is .12 and the p.value is 0.035. :astonished: 

Looks like the p-value refers to the unstandardized coefficient:

```
tab_model(thomas, show.stat = T)
```

![model1](https://user-images.githubusercontent.com/32065313/84896918-18c6de00-b0a5-11ea-8d64-6e0297a9da9d.png)

Unfortunately I cannot share the data as it is not my own. I would assume this should be reproducible with another example, but the problem might just be conceptual. As far as I understand, using `show.std = T` produces a different model by standardizing variables before calling `lmer` (transformations will change estimates in mixed models). Should there not be two t-values and two p-values?

I am using version 2.84 of sjPlot and R 4.0.1

**Comment by strengejacke** [comment]
Can you please post the output from following code:

```r
parameters::model_parameters(thomas)
parameters::model_parameters(thomas, standardize = "refit")
```

**Comment by johannes-titz** [comment]
Wow, that was fast! Sure thing:

```
parameters::model_parameters(thomas)
```
<table>
<thead>
<tr class="header">
<th align="left">Parameter</th>
<th align="right">Coefficient</th>
<th align="right">SE</th>
<th align="right">CI_low</th>
<th align="right">CI_high</th>
<th align="right">t</th>
<th align="right">df_error</th>
<th align="right">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">2.2853535</td>
<td align="right">0.4734923</td>
<td align="right">1.3573255</td>
<td align="right">3.2133814</td>
<td align="right">4.826590</td>
<td align="right">226</td>
<td align="right">0.0000014</td>
</tr>
<tr class="even">
<td align="left">emotion</td>
<td align="right">0.6376902</td>
<td align="right">0.0940350</td>
<td align="right">0.4533851</td>
<td align="right">0.8219954</td>
<td align="right">6.781415</td>
<td align="right">226</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">age</td>
<td align="right">-0.0202625</td>
<td align="right">0.0096182</td>
<td align="right">-0.0391139</td>
<td align="right">-0.0014112</td>
<td align="right">-2.106686</td>
<td align="right">226</td>
<td align="right">0.0351448</td>
</tr>
<tr class="even">
<td align="left">emotion:age</td>
<td align="right">0.0043786</td>
<td align="right">0.0019486</td>
<td align="right">0.0005595</td>
<td align="right">0.0081977</td>
<td align="right">2.247083</td>
<td align="right">226</td>
<td align="right">0.0246347</td>
</tr>
</tbody>
</table>


```
parameters::model_parameters(thomas, standardize = "refit")
```
<table>
<thead>
<tr class="header">
<th align="left">Parameter</th>
<th align="right">Coefficient</th>
<th align="right">SE</th>
<th align="right">CI_low</th>
<th align="right">CI_high</th>
<th align="right">t</th>
<th align="right">df_error</th>
<th align="right">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">0.0088409</td>
<td align="right">0.0338483</td>
<td align="right">-0.0575004</td>
<td align="right">0.0751823</td>
<td align="right">0.2611933</td>
<td align="right">226</td>
<td align="right">0.7939434</td>
</tr>
<tr class="even">
<td align="left">emotion</td>
<td align="right">0.8559836</td>
<td align="right">0.0392252</td>
<td align="right">0.7791037</td>
<td align="right">0.9328635</td>
<td align="right">21.8223062</td>
<td align="right">226</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">age</td>
<td align="right">0.0039523</td>
<td align="right">0.0341124</td>
<td align="right">-0.0629067</td>
<td align="right">0.0708114</td>
<td align="right">0.1158623</td>
<td align="right">226</td>
<td align="right">0.9077617</td>
</tr>
<tr class="even">
<td align="left">emotion:age</td>
<td align="right">0.0888689</td>
<td align="right">0.0395493</td>
<td align="right">0.0113538</td>
<td align="right">0.1663840</td>
<td align="right">2.2470436</td>
<td align="right">226</td>
<td align="right">0.0246372</td>
</tr>
</tbody>
</table>

Sorry again for not being able to provide the raw data.

**Comment by strengejacke** [comment]
That's enough to find out where the problem lies. Indeed, information on standardized estimates, SE and CI are added as new columns, but not the related statistic and p-column.

**Comment by johannes-titz** [comment]
I think the problem is that the t-statistic of the standardized coefficients is not removed before adding the columns to the table (while the p-value is). The fix then is easy:

```diff
@@ -539,7 +539,7 @@ tab_model <- function(
             conf.low = "std.conf.low",
             conf.high = "std.conf.high"
           ) %>%
-          dplyr::select(-1, -.data$p.value) %>%
+          dplyr::select(-1, -.data$p.value, -.data$statistic) %>%
           sjmisc::add_columns(dat) %>%
           dplyr::mutate(std.conf.int = sprintf(
             "%.*f%s%.*f",
```
https://github.com/strengejacke/sjPlot/compare/master...johannes-titz:std

And I get the correct result:

```
tab_model(thomas, show.stat = T, show.std = T)
```

![after_fix](https://user-images.githubusercontent.com/32065313/85856152-c168f200-b7b7-11ea-914f-28b8ad544c43.png)

I ran Check in R and get no errors, warnings or notes. Maybe you want to pull it after testing it yourself?

I am not really sure how bad this issue actually is. Maybe you would like to provide some information in the readme so that people can check if their analyses were/are affected? As far as I understand, every time `show.std = TRUE` there is no guarantee that the p-value and t-value will refer to the same model, at least not for a mixed model. But maybe I am wrong since somebody would have likely noticed this before.

**Comment by strengejacke** [comment]
We could add test statistic and p-values for both standardized and non-standardized estimates.

**Comment by johannes-titz** [comment]
This is a good option, too, but requires more changes. I just tried it and there are several locations where the code doubles in lines because the transformations and formatting have to be applied to both the p-value and the standardized p-value. It works, but not sure if this is the optimal solution. I guess one alternative is to externalize the p-value manipulations into a separate function (emphasize p-value, indicate p <0.001, scientific notation) and then call it with the p-value and the standardized p-value. An advantage of this approach would be that this function could be used elsewhere too and the `tab_model` function will become a bit shorter.

One could also scale the variables of the model first and then call `tab_model` (or parts of it), rename the variables and merge them with the unstandardized model. But I actually like the simplicity of standardizing via `model_parameters`.

What do you think? I could give the first alternative a shot if you like.

**Comment by johannes-titz** [comment]
So I tried to implement the alternative and it seems to work. But I had to change many things so some more tests are required. For instance, I check if t-values are equivalent and if they are, I drop the std.p.value and std.statistic column. Otherwise, for normal linear models two columns are shown, although they are identical. Not sure if I have overseen anything else that might cause issues similar to this one. But *check* does not result in any errors or warnings.

There is now a new function to format the p-value (see last post). 

https://github.com/johannes-titz/sjPlot/tree/stdB

![after_fix2](https://user-images.githubusercontent.com/32065313/85952040-0c753780-b967-11ea-8cc6-cce192239547.png)

**Comment by strengejacke** [comment]
Thanks a lot! I'll look into it, but I guess I can do this first in August...

**Comment by strengejacke** [comment]
Could you please open a PR?

**Comment by strengejacke** [comment]
And add yourself as contributor to DESCRIPTION, if you like

**Comment by johannes-titz** [comment]
The first quick fix would be ok to pull, but the second alternative will need more testing first. I am also not sure how to treat this issue conceptually. Maybe I will have time until August to do some more tests and then you can just pull it. Otherwise you could also run some more tests when you have the time and make the appropriate changes.

**Comment by strengejacke** [comment]
Sounds good! Let's respond here if one of us can actually start looking into the changes more in detail.

**Comment by johannes-titz** [comment]
So I actually freed up some time today :) and ran a couple of simple tests. There was a minor problem with the *show* parameters, which I already fixed. Since I made some bigger changes to the code I wanted to know if I can still reproduce the old results. So I simply recreated the vignettes and compared them. Now, the original problem I described in this issue is already visible in the vignettes: for *tab_model_estimate*, see the table table in section *Adding columns*, for which the t-value of the Intercept is -0.09 and the p-value is <0.001 -- exactly the same problem as with the dataset I initially posted. With my fix the t-value is 18.62. This is the only change in the vignette *tab_model_estimates*, so I assume that my changes did not break anything. For the vignette *tab_robust* the p-values for the standardized coefficients deviate from the p-values for the unstandardized coefficient several times (see table 5 and the last table). *tab_bayes* produces minor differences between some random effects parameters, but I am not really sure why. Is this expected? *tab_mixed* does not produce any differences. If you have an idea what happens in *tab_bayes*, I think this should be good to go. I already added my ctb infos and will open a PR in a second. Still, you should check that everything is the way you want it to be.
